src/plugins/expression_reveal_image/public/components/reveal_image_component.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useRef, useState, useEffect, useCallback } from 'react';
+import { useResizeObserver } from '@elastic/eui';
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { NodeDimensions, RevealImageRendererConfig, OriginString } from '../../common/types';
+import { isValidUrl, elasticOutline } from '../../../presentation_util/public';
+import './reveal_image.scss';
+
+interface RevealImageComponentProps extends RevealImageRendererConfig {
+  handlers: IInterpreterRenderHandlers;
+  parentNode: HTMLElement;
+}
+
+interface ImageStyles {
+  width?: string;
+  height?: string;
+  clipPath?: string;
+}
+
+interface AlignerStyles {
+  backgroundImage?: string;
+}
+
+function RevealImageComponent({
+  handlers,
+  parentNode,
+  percent,
+  origin,
+  image,
+  emptyImage,
+}: RevealImageComponentProps) {
+  const [loaded, setLoaded] = useState(false);
+  const [dimensions, setDimensions] = useState<NodeDimensions>({
+    width: 1,
+    height: 1,
+  });
+
+  const imgRef = useRef<HTMLImageElement>(null);
+
+  const parentNodeDimensions = useResizeObserver(parentNode);
+
+  // modify the top-level container class
+  parentNode.className = 'revealImage';
+
+  // set up the overlay image
+  const updateImageView = useCallback(() => {
+    if (imgRef.current) {
+      setDimensions({
+        height: imgRef.current.naturalHeight,
+        width: imgRef.current.naturalWidth,
+      });
+
+      setLoaded(true);
+      handlers.done();
```

Comment:
nit: seems this is the only handler used now, so i am also in favor of what vadim partialy suggested as well, to not pass whole handlers to this component, but rather just pass the done handler: `<image_reveal onLoaded={handlers.done}` which might make it easier to spot which handlers specific renderer depends upon.

Reply:
Are we sure, that all further components will be using `handlers` in such a simple way? If not, we will provide a bad pattern. What would you say about it?

Reply:
no strong opinion on it, i think most renderers are not gonna use a lot of handlers, but some of course will.

Reply:
Done.